src/main/java/org/apache/sling/xss/impl/AntiSamyPolicyAdapter.java

```diff
@@ -0,0 +1,289 @@
+                        policyBuilder.allowAttributes(CssValidator.STYLE_ATTRIBUTE_NAME)
+                                .matching(cssValidator.newCssAttributePolicy()).onElements(tag.getValue().getName());
+                    }
+                    break;
+
+                default:
+                    throw new RuntimeException("No tag action found.");
+            }
+        }
+
+        // disallow style tag on specific elements
+        policyBuilder.disallowAttributes(CssValidator.STYLE_ATTRIBUTE_NAME)
+                .onElements(cssValidator.getDisallowedTagNames().toArray(new String[0]));
+
+        // ---------- dynamic attributes ------------
+        Map<String, Attribute> dynamicAttributes = new HashMap<>();
+
+        // checks if the dynamic attributes are allowed
+        if (policy.getDirectives().get(ALLOW_DYNAMIC_ATTRIBUTES).equals("true")) {
+            dynamicAttributes.putAll(policy.getDynamicAttributes());
+            for (Attribute attribute : dynamicAttributes.values()) {
+                if (attribute.getOnInvalid().equals(REMOVE_TAG_ON_INVALID_ACTION)) {
+                    onInvalidRemoveTagList.add(attribute.getName());
+                }
+
+                List<Pattern> regexsFromAttribute = attribute.getPatternList();
+                for (Pattern regex : regexsFromAttribute) {
+                    dynamicAttributesPolicyMap.put(attribute.getName(), newDynamicAttributePolicy(regex));
+                }
+
+                List<String> allowedValuesFromAttribute = attribute.getLiterals();
+                if (!allowedValuesFromAttribute.isEmpty()) {
+                    dynamicAttributesPolicyMap.put(attribute.getName(),
+                            newDynamicAttributePolicy(true, allowedValuesFromAttribute.toArray(new String[0])));
+                }
+
+            }
+        }
+
+        policyFactory = policyBuilder.allowTextIn(CssValidator.STYLE_TAG_NAME).toFactory();
+
+    }
+
+    public PolicyFactory getHtmlCleanerPolicyFactory() {
+        return policyFactory;
+    }
+
+    public Map<String, AttributePolicy> getDynamicAttributesPolicyMap() {
+        return dynamicAttributesPolicyMap;
+    }
+
+    public List<String> getOnInvalidRemoveTagList() {
+        return onInvalidRemoveTagList;
+    }
+
+    public CssValidator getCssValidator() {
+        return cssValidator;
+    }
+
+    private static Predicate<String> matchesToPatterns(List<Pattern> patternList) {
+        return new Predicate<String>() {
+            @Override
+            public boolean apply(String s) {
+                for (Pattern pattern : patternList) {
+                    if (pattern.matcher(s).matches()) {
+                        return true;
+                    }
+                }
+                return false;
+            }
+        };
+    }
+
+    private static Predicate<String> matchesToPatternsAndLiterals(List<Pattern> patternList, boolean ignoreCase, List<String> literalList) {
```

Comment:
`matchesPatternsOrLiterals`. Reuse the pattern matcher from above and combine with `or` with a literal matcher to reduce duplication